timesync/peersync/sync_test.go

```diff
@@ -119,16 +119,7 @@ func TestSyncTerminateOnError(t *testing.T) {
 	require.NoError(t, mesh.ConnectAllButSelf())
 	sync.Start()
 	t.Cleanup(sync.Stop)
-	errors := make(chan error, 1)
-	go func() {
-		errors <- sync.Wait()
-	}()
-	select {
-	case err := <-errors:
-		require.ErrorIs(t, err, errPeersNotSynced)
-	case <-time.After(100 * time.Millisecond):
-		require.FailNow(t, "timed out waiting for sync to fail")
-	}
+	require.ErrorIs(t, sync.Wait(), errPeersNotSynced)
```

Comment:
I guess the rationale for that complicated way was not to block the whole test suite if `sync.Wait()` hangs for some reason. But either way is fine for me